test/e2e_node/security_context_test.go

```diff
@@ -114,4 +115,75 @@ var _ = framework.KubeDescribe("Security Context", func() {
 			}
 		})
 	})
+
+	Context("when creating a pod in the host network namespace", func() {
+		makeHostNetworkPod := func(podName, image string, command []string, hostNetwork bool) *v1.Pod {
+			return &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: podName,
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					HostNetwork:   hostNetwork,
+					Containers: []v1.Container{
+						{
+							Image:   image,
+							Name:    podName,
+							Command: command,
+						},
+					},
+				},
+			}
+		}
+		listNetworkInterfaceCommand := []string{"sh", "-c", "awk '{print $1}' /proc/net/dev"}
+		createAndWaitHostNetworkPod := func(podName string, hostNetwork bool) {
+			podClient.Create(makeHostNetworkPod(podName,
+				"gcr.io/google_containers/busybox:1.24",
+				listNetworkInterfaceCommand,
+				hostNetwork,
+			))
+
+			podClient.WaitForSuccess(podName, framework.PodStartTimeout)
+		}
+
+		networkInterfaces := ""
+		BeforeEach(func() {
+			output, err := exec.Command(listNetworkInterfaceCommand[0], listNetworkInterfaceCommand[1:]...).Output()
+			if err != nil {
+				framework.Failf("Execute command %q failed: %v", listNetworkInterfaceCommand, err)
+			}
+			networkInterfaces = strings.TrimSpace(string(output))
+			framework.Logf("Got host network interfaces %q", networkInterfaces)
+		})
+
+		It("should show same network interfaces in the host network containers", func() {
```

Comment:
It's possible that other pods were created after `BeforeEach` but before creating this test pod. If that's the case, won't the test pod see more interfaces? 

Besides running this in a serial suite, are there more robust way to check the host network setting?